src/transport/SecureSession.h

```diff
@@ -60,26 +63,20 @@ class SecureSession : public Session
     {
         kPASE = 1,
         kCASE = 2,
-        // kPending denotes a secure session object that is internally
-        // reserved by the stack before and during session establishment.
-        //
-        // Although the stack can tolerate eviction of these (releasing one
-        // out from under the holder would exhibit as CHIP_ERROR_INCORRECT_STATE
-        // during CASE or PASE), intent is that we should not and would leave
-        // these untouched until CASE or PASE complete.
-        kPending = 3,
     };
 
-    // TODO: This constructor should be private.  Tests should allocate a
-    // kPending session and then call Activate(), just like non-test code does.
-    SecureSession(Type secureSessionType, uint16_t localSessionId, NodeId localNodeId, NodeId peerNodeId, CATValues peerCATs,
-                  uint16_t peerSessionId, FabricIndex fabric, const ReliableMessageProtocolConfig & config) :
-        mSecureSessionType(secureSessionType),
-        mLocalSessionId(localSessionId), mLocalNodeId(localNodeId), mPeerNodeId(peerNodeId), mPeerCATs(peerCATs),
-        mPeerSessionId(peerSessionId), mLastActivityTime(System::SystemClock().GetMonotonicTimestamp()),
-        mLastPeerActivityTime(System::SystemClock().GetMonotonicTimestamp()), mMRPConfig(config)
+    // Test-only: inject a session in Active state.
```

Comment:
Why was the TODO about making tests take the normal codepath removed?  They should still take the normal codepath...